examples/k8s-upgrade/README.md

```diff
@@ -2,18 +2,116 @@
 
 ## Overview
 
-This document describes how to upgrade kubernetes version on a running cluster.
+This document describes how to upgrade kubernetes version for an existing cluster.
 
-Supported scenarios:
-- upgrade from v1.5.x to the latest supported version in v1.6 stream
-- upgrade from v1.6.x to the latest supported version in v1.7 stream
-- upgrade from v1.7.x to the latest supported version in v1.8 stream
+*acs-engine* supports Kubernetes version upgrades starting from 1.5 release.
+During the upgrade, *acs-engine* successively visits virtual machines that constitute the cluster and performs the following operations:
+ - cordon the node and drain existing workload
+ - delete the VM
+ - create new VM and install desired orchestrator version
+ - add the new VM to the cluster
 
-The cluster definition file examples demonstrate initial cluster configurations:
-- **kubernetes1.5.json** - Kubernetes cluster v1.5 with Linux agent pool
-- **kubernetes1.5-win.json** - Kubernetes cluster v1.5 with Windows agent pool
-- **kubernetes1.5-hybrid.json** - Kubernetes cluster v1.5 with Linux and Windows agent pools
-- **kubernetes1.6.json** - Kubernetes cluster v1.6 with Linux agent pool
-- **kubernetes1.7.json** - Kubernetes cluster v1.7 with Linux agent pool
+*acs-engine* allows a subsequent minor version upgrade at a time, for example, from 1.5.x to 1.6.x.
 
-The ***.env** files are used to set desired kubernetes version and instruct test framework to invoke post-deploy instructions implemented in **k8s-upgrade.sh** script.
+For upgrade that spans over more than a single minor version, this operation should be called several times, each time advancing the minor version by one. For example, to upgrade from 1.6.* to 1.8.* one should first upgrade the cluster to 1.7.*, following by upgrading it to 1.8.*

```

Comment:
grammar: followed by upgrading it to 1.8.*
minor improvement: can you put code block markdown around x.y.z for better readability?